torch/lib/c10d/NCCLUtils.cpp

```diff
@@ -0,0 +1,32 @@
+#include <c10d/NCCLUtils.hpp>
+#include <mutex>
+
+namespace c10d {
+
+std::once_flag ncclGetVersionFlag;
+
+std::string getNcclVersion() {
+  std::string versionString;
```

Comment:
This needs to be static or you'll return an empty string on the second call.

You can also move the `once_flag` into this function as a static to isolate it.